script.js

```diff
@@ -1,4 +1,173 @@
+    rolledMessage.push(
+      ` <strong><u>${diceRoll[d]}</u></strong> for Dice ${d + 1}`
+    );
+  }
+  return `${welcomeMessgae} <br>You rolled${rolledMessage} <br>Click Submit to get the combined number.`;
+};
+
+var validateSelection = function (selectedNum) {
+  return selectedNum == 1 || selectedNum == 2;
+};
+
+var validateNumDices = function (selectedNumDice) {
+  if (selectedNumDice != 0 && selectedNumDice != 1) {
+    return selectedNumDice == selectedNumDice;
+  }
+};
+
+var arrayToNum = function (num1, num2) {
+  return Number(String(num1) + String(num2));
+};
+
+var sortArray = function () {
+  if (gameMode == highestCombinedNumMode) {
+    diceRoll = diceRoll.sort(function (a, b) {
+      return b - a;
+    });
+    return diceRoll;
+  } else {
+    return diceRoll.sort();
+  }
+};
+
+var getPlayersNumAuto = function () {
+  diceRoll = sortArray();
+  if (currPlayer == 1) {
+    while (diceRoll.length > 0) {
+      player1Num = arrayToNum(player1Num, diceRoll.shift());
+    }
+    stageMode = diceRollMode;
+    currPlayer = 2;
+    return `Player 1 <br>Your number is ${player1Num}.<br><br>It is now Player 2's turn. <br>Click Submit to roll dice`;
+  }
+  if (currPlayer == 2) {
+    while (diceRoll.length > 0) {
+      player2Num = arrayToNum(player2Num, diceRoll.shift());
+    }
+    stageMode = winnerMode;
+    return `Player 2 <br>Your number is ${player2Num}.<br><br>Click Submit to view winner!!`;
+  }
+};
+
+var getLeaderBoard = function () {
+  player1Score += player1Num;
+  player2Score += player2Num;
+  if (gameMode == highestCombinedNumMode) {
+    if (player1Score > player2Score) {
+      return `Leaderboard: <br>Player 1: ${player1Score} <br>Player 2: ${player2Score}`;
+    }
+    return `Leaderboard: <br>Player 2: ${player2Score} <br>Player 1: ${player1Score}`;
+  }
+  if (gameMode == lowestCombinedNumMode) {
+    if (player1Score > player2Score) {
+      return `Leaderboard: <br>Player 2: ${player2Score} <br>Player 1: ${player1Score}`;
+    }
+    return `Leaderboard: <br>Player 1: ${player1Score} <br>Player 2: ${player2Score}`;
+  }
+};
+
+var getWinner = function () {
+  var genericMessage = `Player 1's number: ${player1Num} | Player 2's number: ${player2Num} <br><br>Click Submit to play again.`;
+  var leaderboard = getLeaderBoard();
+  if (gameMode == highestCombinedNumMode) {
+    if (player1Num > player2Num) {
+      return `Player 1 has won. <br>${genericMessage} <br><br>${leaderboard}`;
+    }
+    return `Player 2 has won. <br>${genericMessage} <br><br>${leaderboard}`;
+  }
+  if (gameMode == lowestCombinedNumMode) {
+    if (player1Num > player2Num) {
+      return `Player 2 has won. <br>${genericMessage} <br><br>${leaderboard}`;
+    }
+    return `Player 1 has won. <br>${genericMessage} <br><br>${leaderboard}`;
+  }
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (!gameMode) {
+    var gameModeSelected = validateSelection(input);
```

Comment:
this validation were initially use to validate the selection of 2 dices and the lowest & highest combined number selection. Since there is a change on the default number of dices, I just keep it for the lowest & highest combined number selection.